### DISCUSSION TITLE
Find closest point in direction, by offset

### DISCUSSION THREAD

**Discussion by GGAlanSmithee** [isaac-mason/recast-navigation-js#250] [Q&A]
Hey, just wanted to ask for input, in case anyone has a good idea.

My usecase is for an entity to travel a path, where the last point is an interactable object, at which the entity should stop at a certain distance from that point, and face it.

I do have some solutions in mind, but wanted to see if there might be a more elegant solution.

Thanks

**Comment by isaac-mason** [comment]
A naive solution could be to check the agent's position each frame (`agent.position()`), and stop the agent (`agent.resetMoveTarget()`) once it's within a given distance from the point.

**Comment by GGAlanSmithee** [comment]
Thanks for your input @isaac-mason 

Yeah, sometimes the easiest solutions are the best one. I think the issues is a bit more nuanced, because you want it to stop in the direction of the last point of movement, which has some edgecases.

I did solve this but projecting outwards from the target and move the agent their instead. I am thinking about using crowds and obstacles instead, that might do this out of the box more or less.

**Reply by isaac-mason** [reply]
Thanks for sharing your solution!

>  I am thinking about using crowds and obstacles instead, that might do this out of the box more or less.

Out of interest, what were you thinking of doing here? Putting an obstacle around the target so the agent can't get closer to it?

**Reply by GGAlanSmithee** [reply]
I have not been using crowds or obstacles in my current solution since I need fine-grained control, and have not yet been able to start looking closer at this, but my naive solution was to do exactly as you say, more or less just put an obstacle around the target, or in case of dynamic targets, make them agents and - if that is even possible - give them a large radius, and if that is not possible, simply handle it in the update loop
